ctapipe/image/muon/muon_ring_finder.py

```diff
@@ -1,28 +1,139 @@
 import numpy as np
 import astropy.units as u
-from ctapipe.image.muon.ring_fitter import RingFitter
+from ctapipe.core import Component
 from ctapipe.io.containers import MuonRingParameter
+import traitlets as traits
+from iminuit import Minuit
+
+__all__ = ['MuonRingFitter']
+
+
+class MuonRingFitter(Component):
+    """Different ring fit algorithms for muon rings
+    """
+    fit_method = traits.CaselessStrEnum(
+        ['taubin', 'chaudhuri_kundu'],
+        default_value='chaudhuri_kundu'
+    ).tag(config=True)
+
+    tel_type = traits.CaselessStrEnum(
```

Comment:
I don't think we should hard code a list of tel-types here. If you require information about the telescopes, the `__call__` method should get the `subarray` and from there you should extract the information you need.
If it's only a few things you need, like the focal length, you should just give that as input.